portfolio/src/main/webapp/script.js

```diff
@@ -39,27 +37,59 @@ function addHobby() {
  */
 function addProjectDescription(project) {
   
-  const details = {
-    nlp: 'Building language models',
-    ta: 'Held weekly office hours and review sessions for introductory Data Structure and Algorithms class',
-    helmet: 'Integrated speech command to a \"smart\" bike helmet built by a team of 9 friends',
-    unknown: 'Something is wrong, project doesn\'t exist'
-  };
-  
-  const colors = {
-    nlp: '#6ccfe0',
-    ta: '#f7d36f',
-    helmet: '#e0b9f0',
-    unknown: 'black'
-  };
+  let projectName = project.toUpperCase(); 
+
+  const projectNameToData = {
+    NLP: createProjectData('Building language models', '#6ccfe0'),
+    TA: createProjectData('Held weekly office hours and review sessions for introductory Data Structure and Algorithms class', '#f7d36f'),
+    HELMET: createProjectData('Integrated speech command to a \"smart\" bike helmet built by a team of 9 friends', '#e0b9f0'),
+    UNKNOWN: createProjectData('Something is wrong, project doesn\'t exist', 'black')
+  }
   
-  if (!(project in details)) {
-    project = 'unknown'; 
+  if (!(projectName in projectNameToData)) {
+    projectName = 'UNKNOWN'; 
   };
   
-  const wrapperName = 'project-description-';
+  const wrapperName = 'projectDescription';
   const wrapper = document.getElementById(wrapperName.concat(project));
   
-  wrapper.innerText = details[project]; 
-  wrapper.style.backgroundColor = colors[project];
+  wrapper.innerText = projectNameToData[projectName]['projectDetail']; 
+  wrapper.style.backgroundColor = projectNameToData[projectName]['backgroundColor']; 
+}
+
+
+/**
+ * This is the helper function for addProjectDescription.
+ * This function creates a data key-value pair object for each project.
+ * Input: 
+ */
+function createProjectData(projectDetail, backgroundColor) {
+ 
+  projectData = {}
+  projectData['projectDetail'] = projectDetail;
+  projectData['backgroundColor'] = backgroundColor; 
+
+  return projectData; 
+}
+
+/**
+ * This function enables the highlighting of a selected section from the navigation bar. 
+ * This function is called by the clicking of any one of the boxes in the navigation bar.
+ * Input: section to select
+ * Effect: selected section will have their border shown, and all the other sections will not have the border. 
+ */
+function highlightSection(selectedIdString) {
+
+  document.getElementById(selectedIdString).style.border = 'solid';
+  document.getElementById(selectedIdString).style.borderColor = '#c5cf0a';
+
+  let navbarSections = document.getElementsByClassName('navbarSection');
+
+  Object.keys(navbarSections).forEach(
+    (key) => {
+      var section = navbarSections[key];
```

Comment:
nit: Prefer const over var or let whenever you can.

Reply:
Done